src/app/shared/components/login-form/login-form.component.ts

```diff
@@ -1,21 +1,24 @@
-import { Component, ViewChild } from '@angular/core';
-import { NgForm } from '@angular/forms';
-import { faEye, faEyeSlash, IconDefinition } from '@fortawesome/free-solid-svg-icons';
+import { Component, ViewChild } from "@angular/core";
+import { NgForm } from "@angular/forms";
+import { ButtonTypes } from "@app/shared/types/button.type";
+import { IconNames } from "@app/shared/types/icons.model";
 
 @Component({
-  selector: 'app-login-form',
-  templateUrl: './login-form.component.html',
-  styleUrls: ['./login-form.component.scss'],
+  selector: "app-login-form",
+  templateUrl: "./login-form.component.html",
+  styleUrls: ["./login-form.component.scss"],
 })
-export class LoginFormComponent{
+export class LoginFormComponent {
+  ButtonTypes = ButtonTypes;
+  IconNames = IconNames;
+
   @ViewChild("loginForm") public loginForm!: NgForm;
   //Use the names `email` and `password` for form controls.
-  email:string ='';
-  password:string='';
-  submitted='false';
-  visible:IconDefinition=faEye;
-  hidden:IconDefinition=faEyeSlash;
-
+  email = "";
+  password = "";
+  submitted = "false";
 
-  onSubmit() {}
+  onSubmit() {
+    return undefined;
```

Comment:
Hmmm... this looks very odd.

Reply:
this is just because of linter error. I dont want to see files marked red only if there is a real problem. linter doesnt like empty functions. -> will be changed when onSubmit has a functionality.